src/hardware/adc_internal.rs

```diff
@@ -0,0 +1,133 @@
+use super::hal::{
+    adc,
+    gpio::{gpioa::*, gpiob::*, gpioc::*, gpiof::*, Analog},
+    hal::blocking::delay::DelayUs,
+    prelude::*,
+    rcc::{rec, CoreClocks},
+    stm32::{ADC1, ADC2, ADC3},
+};
+
+pub enum TecChannel {
+    Zero,
+    One,
+    Two,
+    Three,
+}
+
+pub enum Supply {
+    P5v,
+    P12v,
+    P3v3,
+    I12v,
+}
+
+pub struct AdcInternal {
+    adc1: adc::Adc<ADC1, adc::Enabled>,
+    adc3: adc::Adc<ADC3, adc::Enabled>,
+    p5v: PC0<Analog>,
```

Comment:
I'd make `SupplyPins` in analogy to `TecUPins` and `TecIPins`.

Also: is `Tec` the right term here and is it/can it/will it be used consistently?

Reply:
I'll name `Tec` `Output`